server/test/controllers/applicant/ApplicantProgramsControllerTest.java

```diff
@@ -254,6 +257,18 @@ public void show_withStringProgramParam_showsByProgramSlug() {
         .contains(routes.ApplicantProgramReviewController.review(program.id).url());
   }
 
+  @Test
+  public void showInfoDisabledProgram() {
+    ProgramModel disabledProgram = resourceCreator.insertActiveDisabledProgram("disabledProgram");
+
+    Map<String, String> flashData = new HashMap<>();
+    flashData.put("redirected-from-program-slug", "disabledProgram");
+    Request request = Helpers.fakeRequest().flash(flashData).build();
+
+    Result result = controller.showInfoDisabledProgram(request).toCompletableFuture().join();
+    assertThat(result.status()).isEqualTo(NOT_FOUND);
```

Comment:
suggestion (non-blocking): you could also assert that `contentAsString(result)` contains the text you expect to see on the page.